include/zephyr/mgmt/mcumgr/grp/os_mgmt/os_mgmt.h

```diff
@@ -98,6 +102,16 @@ struct os_mgmt_info_append {
 	bool *prior_output;
 };
 
+/* MCUboot mode of operation */
+enum os_mgmt_mcuboot_info_mode {
```

Comment:
This can go, it can use the mcuboot defines directly (though mcuboot needs to be updated to add a new one for XIP with revert)

Reply:
I can not because the defines need to be hardcoded, as protocol definition, and documentation, requires exact values.
Additionally the MCUboot defines have the UPGRADE only mode, which is actually variant of swap or swap with scratch, without distinction, while the modes here provide different codes for both modes and extra flags when any of them is paired with downgrade prevention.

Reply:
The numbers are fixed though, in mcuboot, new numbers will be appended if there are new modes going forward (XIP with revert got inserted because the original PR was only just merged and isn't being used). This needs to support showing if the mode is upgrade only, it is wrong for MCUboot to be configured in upgrade mode only and for this to show it is a swap with/without scratch mode.